jena-fuseki2/jena-fuseki-ui/src/components/dataset/Pagination.vue

```diff
@@ -16,110 +16,124 @@
-            type="button"
-            role="menuitem"
-            aria-label="Go to last page"
-          >
-            ›
-          </button>
-        </li>
-        <li
-          :aria-hidden="getNextLinkAriaDisabled()"
-          :class="getNextLinkClass('page-item')"
-          role="presentation"
-        >
-          <button
-            :aria-disabled="getNextLinkAriaDisabled()"
-            :class="getNextLinkClass('page-link')"
-            @click="goToPage(numberOfPages)"
-            type="button"
-            role="menuitem"
-            aria-label="Go to next page"
-          >
-            »
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
+        «
+      </button>
+    </li>
+    <li
+      :aria-hidden="isBackLinkAriaDisabled"
+      :class="getBackLinkClass('page-item')"
+      role="presentation"
+    >
+      <button
+        :aria-disabled="isBackLinkAriaDisabled"
+        :class="getBackLinkClass('page-link')"
+        @click="goToPage(currentPage - 1)"
+        type="button"
+        aria-label="Go to previous page"
+        role="menuitem"
+      >
+        ‹
+      </button>
+    </li>
+    <li
+      v-if="isBackSummaryEnabled"
+      role="presentation"
+      class="page-link page-summary-item"
+    >
+      <span>…</span>
+    </li>
+    <!-- pages -->
+    <li
+      v-for="page in pages"
+      :key="page"
+      role="presentation"
+      class="page-item"
+    >
+      <span
+        :aria-label="`Go to page ${ page }`"
+        :class="getPageLinkClass(page)"
+        :aria-checked="page === currentPage"
+        @click="goToPage(page)"
+        role="menuitemradio"
+        type="button"
+        aria-posinset="1"
+        aria-setsize="2"
+        tabindex="0"
+      >{{ page }}</span>
+    </li>
+    <!-- pages forward controls -->
+    <li
+      v-if="isAfterSummaryEnabled"
+      role="presentation"
+      class="page-link page-summary-item"
+    >
+      <span>…</span>
+    </li>
+    <li
+      :aria-hidden="isNextLinkAriaDisabled"
+      :class="getNextLinkClass('page-item')"
+      role="presentation"
+    >
+      <button
+        :aria-disabled="isNextLinkAriaDisabled"
+        :class="getNextLinkClass('page-link')"
+        @click="goToPage(currentPage + 1)"
+        type="button"
+        role="menuitem"
+        aria-label="Go to last page"
+      >
+        ›
+      </button>
+    </li>
+    <li
+      :aria-hidden="isNextLinkAriaDisabled"
+      :class="getNextLinkClass('page-item')"
+      role="presentation"
+    >
+      <button
+        :aria-disabled="isNextLinkAriaDisabled"
+        :class="getNextLinkClass('page-link')"
+        @click="goToPage(numberOfPages)"
+        type="button"
+        role="menuitem"
+        aria-label="Go to next page"
+      >
+        »
+      </button>
+    </li>
+  </ul>
 </template>
 
 <script>
+function range (size, startAt) {
+  return [...Array(size).keys()].map(i => i + startAt)
+}
```

Comment:
![image](https://github.com/apache/jena/assets/304786/c4788765-b0a0-4820-ad66-ef92a8f0724d)

Similar to Python's `range` function. If other components need it, this can be moved to a separate module and shared by them. For now keeping here in the `Pagination` component to simplify it.